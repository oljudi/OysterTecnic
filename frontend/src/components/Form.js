import React from 'react';
import { Box, Stack, Heading, Button } from '@chakra-ui/core';

const Form = ({width, children, title, submit, bgColor, titleButton}) => {
    return (
        <Box
        backgroundColor={bgColor}
        onSubmit={submit}
        as="form"
        w={width || "350px"}
        boxShadow="x1"
      >
        <Stack spacing={8} p={8}>
          <Heading textAlign="center" as="h1" color="black">
            {title}
          </Heading>
          {children}
          <Button backgroundColor="#20D7AE" type="submit" color="white">
            {titleButton}
          </Button>
        </Stack>
      </Box>
    );
};

export default Form;