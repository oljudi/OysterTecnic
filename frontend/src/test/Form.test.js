import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';

import Form from "../components/Form";

describe('Test component <From /> for login', () => {

    const login = e => {}

    const wrapper = shallow(
        <Form
            submit={login}
            bgColor="transparent" 
            titleButton="Acceder!" 
            title="Bienvenido"
        >

        </Form>
        )

    test('Should render <From />', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Should have tittles', () => {
        const tittle = wrapper.find('#tittle').text()
        expect(tittle).toBe("Bienvenido")
        const tittleButton = wrapper.find('#button').text()
        expect(tittleButton).toBe("Acceder!")
    })
})