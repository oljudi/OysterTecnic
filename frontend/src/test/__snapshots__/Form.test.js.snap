// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <From /> for login Should render <From /> 1`] = `
<Box
  as="form"
  backgroundColor="transparent"
  boxShadow="x1"
  onSubmit={[Function]}
  w="350px"
>
  <ForwardRef
    p={8}
    spacing={8}
  >
    <Heading
      as="h1"
      color="black"
      id="tittle"
      textAlign="center"
    >
      Bienvenido
    </Heading>
    <Button
      backgroundColor="#20D7AE"
      color="white"
      id="button"
      type="submit"
    >
      Acceder!
    </Button>
  </ForwardRef>
</Box>
`;
